File: 5Programe.cs
using System;
using System.Collections.Generic;
using System.Linq;

class _5Programe
{
    private static string citesteCLI(string[] args) {
        //// 
        //// citeste din linia de comanda numele fisierului
        //// 
        string stringCLI = "";
        for (int i = 0; i < args.Length; i++) {
            stringCLI += args[i] + " ";
        }
        stringCLI.Trim();
        return stringCLI;
    }

    private static List<List<string>> citesteFisier(string numeFisier) {
        //// 
        //// citire a matricei dintr-un fisier
        //// 
        List<List<string>> matrice = new List<List<string>>();
        
        string[] linii = System.IO.File.ReadAllLines(numeFisier);
        foreach(string linie in linii) {
            List<string> linieArr = new List<string>(linie.Split(new string[] { " " }, StringSplitOptions.None));
            matrice.Add(linieArr);
        }
        return matrice;
    }

    private static void arataMatrice(List<List<string>> matrice) {
        //// 
        //// afisare a matrice linie cu linie
        //// 
        foreach (List<string> i in matrice)
        {
            string linie = "";
            foreach (string j in i)
            {
                linie += j + " ";
            }
            linie.Trim();
            Console.WriteLine(linie);
        }
    }

    private static void arataMatriceElemente(List<List<string>> matrice) {
        //// 
        //// afisare a matrice element cu element dupa forma nr linie, nr coloana si valoare
        //// 
        for (int i = 0; i < matrice.Count; i++)
        {
            for (int j = 0; j < matrice[i].Count; j++)
            {
                Console.WriteLine("Linie " + i + " | Coloana " + j + " | Valoare " + matrice[i][j]);
            }
        }
    }

    private static int verificaVector(List<string> vector1, List<string> vector2) {
        //// 
        //// functie ajutatoare pentru sortarea matricei
        //// 
        int n = 0;
        if (vector1.Count < vector2.Count) {
            n = vector1.Count;
        }
        else n = vector2.Count;
        for (int i = 0; i < n; i++)
        {
            if (String.Compare(vector1[i], vector2[i]) > 0) {
                return 1;
            }
            else if (String.Compare(vector1[i], vector2[i]) < 0) {
                return -1;
            }
        }
        return 0;
    }

    private static void sorteazaMatrice(List<List<string>> matrice) {
        //// 
        //// sorteaza matrice data prin tehnica sortarii prin insertie
        //// 
        for (int i = 0; i < matrice.Count; i++)
        {
            List<string> cheie = matrice[i];
            int j = i - 1;

            while (j >= 0 && verificaVector(cheie, matrice[j]) < 0) {
                matrice[j + 1] = matrice[j];
                j--;
            }
            matrice[j + 1] = cheie;
        }
    }

    static void Main(string[] args)
    {
        //se da fisier in linia de comanda
        string stringInput = citesteCLI(args);
        
        //citire matrice din fisier
        List<List<string>> matriceCitita = citesteFisier(stringInput);

        //afisare intreaga matrice
        Console.WriteLine("Inainte de sortare: ");
        arataMatrice(matriceCitita);

        //afisare matrice element cu element
        arataMatriceElemente(matriceCitita);
        
        //sortam matricea folosing sortarea prin insertie
        sorteazaMatrice(matriceCitita);
        
        //afisam la final matricea
        Console.WriteLine("---------------------\nDupa sortare: ");
        arataMatrice(matriceCitita);
    }
}

File: 5Programe.php
<?php

    function citesteCLI($argv, $argc) {
        /**
         * citeste din linia de comanda numele fisierului
         */
        $stringCLI = "";
        for ($i=1; $i < $argc; $i++) {
            $stringCLI .= $argv[$i] . ' ';
        }
        $stringCLI = trim($stringCLI);
        return $stringCLI;
    }

    function citesteFisier($numeFisier) {
        /**
         * citire a matricei dintr-un fisier
         */
        $matrice = [];

        $linii = file($numeFisier);
        foreach ($linii as $linie) {
            $matrice[] = explode(' ', $linie);
        }
        return $matrice;
    }

    function arataMatrice($matrice) {
        /**
         * afisare a matrice linie cu linie
         */
        for($i = 0; $i < count($matrice); $i++) {
            $linie = '';
            for ($j = 0; $j < count($matrice[$i]); $j++) {
                $linie .= ' ' . $matrice[$i][$j];
            }
            $linie = trim($linie);
            $linie .= "\n";
            echo $linie;
        }
    }

    function arataMatriceElemente($matrice) {
        /**
         * afisare a matrice element cu element dupa forma nr linie, nr coloana si valoare
         */
        for ($i = 0; $i < count($matrice); $i++) {
            for ($j = 0; $j < count($matrice[$i]); $j++) {
                echo 'Linie ' . $i . ' | Coloana ' . $j . ' | Valoare ' . $matrice[$i][$j] . "\n";
            }
        }
    }

    function verificaVector($vector1, $vector2) {
        /**
         * functie ajutatoare pentru sortarea matricei
         */
        $n = 0;
        if (count($vector1) < count($vector2)) {
            $n = count($vector1);
        }
        else $n = count($vector2);
        for ($i = 0; $i < $n; $i++) {
            if ($vector1[$i] > $vector2[$i]) {
                return 1;
            } elseif ($vector1[$i] < $vector2[$i]) {
                return -1;
            }
        }
        return 0;
    }

    function sorteazaMatrice(&$matrice) {
        /**
         * sorteaza matrice data prin tehnica sortarii prin insertie
         */
        for ($i = 0; $i < count($matrice); $i++) {
            $cheie = $matrice[$i];
            $j = $i - 1;

            while ($j >= 0 && verificaVector($cheie, $matrice[$j]) < 0) {
                $matrice[$j + 1] = $matrice[$j];
                $j--;
            }
            $matrice[$j + 1] = $cheie;
        }
    }

    function startapp($argv, $argc) {
        //se da fisier in linia de comanda
        $stringInput = citesteCLI($argv, $argc);
        
        //citire matrice din fisier
        $matriceCitita = citesteFisier($stringInput);
        
        //afisare intreaga matrice
        echo "Inainte de sortare: \n";
        arataMatrice($matriceCitita);
        
        //afisare matrice element cu element
        arataMatriceElemente($matriceCitita);
        
        //sortam matricea folosing sortarea prin insertie
        sorteazaMatrice($matriceCitita);
        
        //afisam la final matricea
        echo "---------------------\nDupa sortare: \n";
        arataMatrice($matriceCitita);
    }

    startapp($argv, $argc);
?>